app/services/reactivate_account_session.rb

```diff
@@ -40,7 +40,7 @@ def validated_personal_key?
   # @return [Pii::Attributes, nil]
   def decrypted_pii
     return unless validated_personal_key?
-    Pii::Cacher.new(@user, session).fetch
+    Pii::Cacher.new(@user, session).fetch(@user.password_reset_profile.id)
```

Comment:
Added the reader.

I'm less happy about removing :x509 in this ticket, because the session is not local to the class. I agree that it doesn't seem to be used anywhere, and we probably ought to remove it, but I prefer opening another ticket, since it's a global change and not directly related to the ticket.

Reply:
Opened [LG-11757](https://cm-jira.usa.gov/browse/LG-11757) to remove the :x509 key.